### DISCUSSION TITLE
运行sssp算法的时候文件sssp.h的Incval函数在哪调用呀

### DISCUSSION THREAD

**Discussion by TerrenceYi** [alibaba/GraphScope#3109] [Q&A] [answered]
sssp.h文件的代码中，Incval函数描述的是Incval阶段一次迭代，我看到basic_graph_benchmarks.cc里面是直接用函数loadandapp运行sssp算法，但是loadandapp的模版中好像没有循环调用Incval函数的代码，求好心人解答

**Comment by doudoubobo** [comment, marked as answer]
在 analytical_engine/core/worker/ 目录下面各种worker调用的，例如在[default worker](https://github.com/alibaba/GraphScope/blob/main/analytical_engine/core/worker/default_worker.h#L120)，当依然有消息产生时，就会循环调用IncEval

**Comment by TerrenceYi** [comment]
> 在 analytical_engine/core/worker/ 目录下面各种worker调用的，例如在[default worker](https://github.com/alibaba/GraphScope/blob/main/analytical_engine/core/worker/default_worker.h#L120)，当依然有消息产生时，就会循环调用IncEval
我找到了，感谢回答
